### PR TITLE
Adding an Automations CLI

### PR DESCRIPTION
**Pull request by chrisguidry**
This CLI supports listing, inspecting, pausing, resuming, and deleting
automations against Cloud, and against the OSS Prefect server when the
experimental flag is on.

Note I'm also consolidating some tests and files to keep as clean of a coverage
target for events as possible while we're still heavily developing on it.

### CODE REVIEW

File: src/prefect/client/orchestration.py [hunk 1]
```diff
@@ -3172,6 +3144,107 @@ async def delete_flow_run_input(self, flow_run_id: UUID, key: str):
...
+            raise RuntimeError("Automations are only supported for Prefect Cloud.")
+
+        response = await self._client.patch(
+            f"/automations/{automation_id}", json={"enabled": False}
+        )
+        response.raise_for_status()
+
+    async def resume_automation(self, automation_id: UUID):
+        if not self.server_type.supports_automations():
+            raise RuntimeError("Automations are only supported for Prefect Cloud.")
```
[the remark this answers is not in the document]
Comment (chrisguidry) [reply src/prefect/client/orchestration.py:3215]: The `supports_automations` method on the enum looks at the flag

[the remark this answers is not in the document]
Comment (zzstoatzz) [reply src/prefect/client/orchestration.py:3215]: is this still a true statement?
> "Automations are only supported for Prefect Cloud."

[the remark this answers is not in the document]
Comment (chrisguidry) [reply src/prefect/client/orchestration.py:3215]: Yes?  Until the experimental flag is removed, right?  I'm not sure how we usually talk about this.  Maybe these should say `This server does not support automations`?

**Review by zzstoatzz — APPROVED** [review]

File: src/prefect/client/orchestration.py [hunk 2]
```diff
@@ -3172,6 +3146,121 @@ async def delete_flow_run_input(self, flow_run_id: UUID, key: str):
...
+                "The current server and client configuration does not support "
+                "events.  Enable experimental events support with the "
+                "PREFECT_EXPERIMENTAL_EVENTS setting."
+            )
+        else:
+            raise RuntimeError(
+                "The current server and client configuration does not support "
+                "automations.  Enable experimental automations with the "
+                "PREFECT_API_SERVICES_TRIGGERS_ENABLED setting."
+            )
```
Comment (chrisguidry) [inline src/prefect/client/orchestration.py:3161]: @zzstoatzz how's this look?

Comment (zzstoatzz) [reply src/prefect/client/orchestration.py:3161]: 💯 🙇

**Pull request merged**